pkg/util/container/ring/buffer_test.go

```diff
@@ -131,3 +134,82 @@ func TestBuffer(t *testing.T) {
 			}
 		})
 }
+
+func TestBufferRandom(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Reference implementation of a queue.
+	var queue []int
+	push := func(x int) {
+		queue = append(queue, x)
+	}
+	pop := func(count int) {
+		require.LessOrEqual(t, count, len(queue))
+		queue = queue[count:]
+	}
+
+	// Validate at each step that the reference queue matches the circular buffer.
+	var b Buffer[int]
+	check := func() {
+		require.Equal(t, b.Length(), len(queue))
+		for i, x := range queue {
+			require.Equal(t, x, b.At(i))
+		}
+	}
+
+	seed := timeutil.Now().UnixNano()
+	t.Logf("Seed: %d", seed)
+	rng := rand.New(rand.NewSource(seed))
+
+	operate := func(pushSkew float64, minSize, maxSize int, popEach int) {
```

Comment:
Could have made it simpler:
- Set a fixed number of steps.
- A fixed prob of push/pop.
- And distribution for pop size.

Would need to make sure the thing does not blow up to a large size, by tuning the params.

LMK which sounds better, or if you have ideas how to stress `Buffer`-specific details more (e.g. the fact that it has a specific policy when to shrink).